conda_smithy/tests/test_lint_recipe.py

```diff
@@ -206,6 +206,38 @@ def test_redundant_license(self):
                             'the word "License".')
         self.assertIn(expected_message, lints)
 
+    def test_end_empty_line(self):
+        bad_contents = [
+            # No empty lines at the end of the file
+            b'extra:\n  recipe-maintainers:\n    - goanpeca',
+            b'extra:\r  recipe-maintainers:\r    - goanpeca',
+            b'extra:\r\n  recipe-maintainers:\r\n    - goanpeca',
+            # Two empty lines at the end of the file
+            b'extra:\n  recipe-maintainers:\n    - goanpeca\n\n',
+            b'extra:\r  recipe-maintainers:\r    - goanpeca\r\r',
+            b'extra:\r\n  recipe-maintainers:\r\n    - goanpeca\r\n\r\n',
+            # Three empty lines at the end of the file
+            b'extra:\n  recipe-maintainers:\n    - goanpeca\n\n\n',
+            b'extra:\r  recipe-maintainers:\r    - goanpeca\r\r\r',
+            b'extra:\r\n  recipe-maintainers:\r\n    - goanpeca\r\n\r\n\r\n',
+        ]
```

Comment:
Sorry, was trying to save you from going down this road by trying to handle unicode cleanly throughout the linter and associated tests. ( https://github.com/conda-forge/conda-smithy/pull/372 )

I think it should be ok to make these explicitly unicode. Though I'll let you be the judge.

The `\r*` tests open an interesting question of should we permit this. 😄 Though I see no harm in allowing it for now and tightening down later. For a related problem, please see issue ( https://github.com/conda-forge/conda-forge-maintenance/issues/8 ).